keyboards/ydkb/mountain/readme.md

```diff
@@ -0,0 +1,217 @@
+
+-   Keyboard Maintainer: [Wraul](https://github.com/Wraul)
+-   Hardware Supported: The YDKB compatible PCB provided as part of the initial
+    batch sold by KBDfans.
+    initial batch. It is most likely also compatible with the hot-swappable PCB,
+    but this is unverified.
+-   Hardware Availability: [KBDfans](https://kbdfans.com/products/ic-kbdfans-mountain-ergo-keyboard-kit)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ydkb/mountain:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+The keyboard comes with a custom Mass Storage Device bootloader and a TMK based firmware from [ydkb.io](https://ydkb.io/).
+Some more information about the bootloader can be found in the [YDKB
+documentation](https://ydkb.io/help/#/en/bootloader/msd-bootloader).
+
+Enter the bootloader in 3 ways:
+
+-   **Bootmagic reset**: Hold down the escape key and plug in the keyboard
+-   **Physical reset button**: The PCB does not have a physical reset button,
+    but it is possible to short the GND and RESET manually. The pins are mapped
+    to two of the unsoldered pads on the controller PCB
+-   **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
+
+### Flashing on Linux
+
+Flashing a firmware can be done on Linux by entering the bootloader and using
+`dd` as described below.
+
+The storage device can differ between systems, so make sure to specify the correct device.
+To list available devices `lsblk` can be used.
+
+```
+dd if=./ydkb_mountain_{LAYOUT}.bin of=/dev/sd{X} seek=4 && sync
+```
+
+When flashing using this method I have at multiple times encountered some form
+of corruption. Once this happens, further flash attempts will silently fail.
+
+I was able to fix the corruption by switching to Windows and doing the flash using
+the file explorer. Make sure to first remove the existing `.bin` file before
+adding the new firmware file to the mounted drive.
+
+## Matrix
+
+The matrix of this keyboard is a bit unusual. Not only is the PCB split into two
+halves, it uses multiple 74HC595 bit shift registers to scan each key
+individually.
+
+### Pins
+
+The following two pins are used to control the 74HC595 bit shift registers.
+
+| ATmega Pin | 74HC595 Pin |
+| ---------- | ----------- |
+| `B1`       | `SRCLK`     |
+| `B3`       | `SER`       |
+
+The following two pins are used to read the state of the matrix.
+
+| ATmega Pin | Function   |
+| ---------- | ---------- |
+| `B2`       | Right half |
+| `B3`       | Left half  |
+
+As shown above the pin `B3` is used both to control the 74HC595 and to read the
+state of the matrix. Thus it has to be switch between output and input modes
+between the two phases.
```

Comment:
I still don't fully understand how this is supposed to work at the schematic level (when some key is pressed, the `B3` pin would be pulled low by the shift register output through the switch when that key is scanned, but at the next step `B3` needs to be forced high in order to write 1 into the shift register).  Or are the SER and key status signals actually separate at the connector, and there is some resistor on the controller board to allow the MCU pin output to override the input from the key?

Reply:
I don't think my electronics skills are enough to understand the exact details. But yes there is a resistor. If I remember correctly it is `R1` in the bottom right of [this picture](https://kbdfans.com/cdn/shop/products/6_02a8c17a-99dd-4540-9cfd-40642542aa0c.jpg?v=1649294337). I think it was in the mega ohm range. And it caused me some grief when I tried to do some continuity probing when figuring out how the keys on the right PCB was scanned.

If you want me to I can check the exact specs and try to figure out exactly how it is connected.